redaxo/src/addons/media_manager/lib/media_manager.php

```diff
@@ -424,6 +424,37 @@ private static function getEffectClass(string $effectFile): string
      * For ExtensionPoints.
      */
 
+    /**
+     * Checks if media is used by this addon.
+     * @param rex_extension_point $ep
+     * @return string[] Warning message as array
+     */
+    public static function mediaIsInUse(rex_extension_point $ep)
+    {
+        /** @var string[] $warning */
+        $warning = $ep->getSubject();
+        $filename = $ep->getParam('filename');
+        assert(is_string($filename));
+
+        $sql = rex_sql::factory();
+        $sql->setQuery('
+            SELECT DISTINCT effect.id AS effect_id, effect.type_id, type.id, type.name
+            FROM `' . rex::getTable('media_manager_type_effect') . '` AS effect
+            LEFT JOIN `' . rex::getTable('media_manager_type') . '` AS type ON effect.type_id = type.id
+            WHERE parameters LIKE ?
+        ', ['%'.$sql->escapeLikeWildcards(json_encode($filename)).'%']);
```

Comment:
Ich habe den Code noch etwas angepasst. Insbesondere habe ich es vom Regex zu LIKE mit prepared Statements geändert.
Da `parameters` JSON ist, können wir davon ausgehen, dass der Filename einzeln in Anführungszeichen irgendwo im JSON sitzt, denke ich.

Reply:
Perfekt. VIelen Dank!